Source/WebCore/platform/graphics/skia/NativeImageSkia.cpp

```diff
@@ -41,6 +40,35 @@ IGNORE_CLANG_WARNINGS_END
 
 namespace WebCore {
 
+void PlatformImageNativeImageBackend::finishAcceleratedRenderingAndCreateFence()
+{
+    auto* glContext = PlatformDisplay::sharedDisplay().skiaGLContext();
+    if (!glContext || !glContext->makeContextCurrent())
+        return;
+
+    auto* grContext = PlatformDisplay::sharedDisplay().skiaGrContext();
+    RELEASE_ASSERT(grContext);
+
+    grContext->flush(m_platformImage);
+
+    if (GLFence::isSupported()) {
+        grContext->submit(GrSyncCpu::kNo);
+        m_fence = GLFence::create();
+    }
+
+    if (!m_fence)
+        grContext->submit(GrSyncCpu::kYes);
```

Comment:
Why does not this function use `flushAndSubmit()` like what `ImageBufferSkiaAcceleratedBackend::finishAcceleratedRenderingAndCreateFence()` does?

Reply:
@alexgcastro asked the same -- Skias API is not orhogonal for SkSurface / SkImage -- the variant that we use for SkSurface doesn't exist for SkImage. You cannot pass the GrSyncCpu option to flushAndSubmit(SkImage) :-(
Therefore it has to be slightly different.